### PR TITLE
chore(litmus-portal): Refactor of Unimodal and Button

### PR DESCRIPTION
**Pull request by saswatamcode**
## Proposed changes
This PR removes the margins set in `ButtonFilled` and `ButtonOutline` components, and applies the required style changes in the components using them. Also includes refactor of `Unimodal` to extend material-ui `ModalProps`, which will make it more modular, and style changes that are required for correct rendering of close button for the modal.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Special notes for your reviewer:
There are no such functional changes.

### CODE REVIEW

**Comment by Saranya-jena** [comment]
@saswatamcode Can attach some screenshots of the modals here?

**Comment by saswatamcode** [comment]
Sure @Saranya-jena!
Screenshot 2020-10-14 at 11 51 19 AM
Screenshot 2020-10-14 at 11 49 21 AM
Screenshot 2020-10-14 at 11 46 46 AM
Screenshot 2020-10-14 at 11 46 25 AM

**Review by Saranya-jena — APPROVED** [review]
LGTM

**Review by Saranya-jena — DISMISSED** [review]

File: litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/index.tsx [hunk 1]
```diff
@@ -182,68 +182,70 @@ const ChooseWorkflow: React.FC = () => {
...
-              >
-                <div>Save</div>
-              </ButtonFilled>
+            <div className={classes.buttons}>
+              <div className={classes.cancelButton}>
+                <ButtonOutline
+                  handleClick={() => setOpen(false)}
+                  isDisabled={false}
+                >
+                  <div>Cancel</div>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/index.tsx:237]:
```suggestion
                  Cancel
```

File: litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/index.tsx [hunk 2]
```diff
@@ -182,68 +182,70 @@ const ChooseWorkflow: React.FC = () => {
...
+                </ButtonOutline>
+              </div>
+              <div className={classes.saveButton}>
+                <ButtonFilled
+                  isPrimary={false}
+                  isDisabled={!isSuccess.current}
+                  handleClick={() => handleSave()}
+                >
+                  <div>Save</div>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/index.tsx:246]:
```suggestion
                  Save
```

File: litmus-portal/frontend/src/views/CreateWorkflow/ReliabilityScore/index.tsx [hunk 3]
```diff
@@ -103,24 +103,29 @@ const ReliablityScore = () => {
...
-                    >
-                      <div>Got it</div>
-                    </ButtonFilled>
-                  </Center>
+                  <div>
+                    <ResultTable
+                      testValue={testWeights}
+                      testNames={testNames}
+                    />
+                  </div>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/CreateWorkflow/ReliabilityScore/index.tsx:116]:
```suggestion
                    <ResultTable
                      testValue={testWeights}
                      testNames={testNames}
                    />
```

File: litmus-portal/frontend/src/views/CreateWorkflow/ReliabilityScore/index.tsx [hunk 4]
```diff
@@ -103,24 +103,29 @@ const ReliablityScore = () => {
...
+                  <hr className={classes.horizontalLineResult} />
+                  <div className={classes.gotItBtn}>
+                    <Center>
+                      <ButtonFilled
+                        handleClick={() => setOpen(false)}
+                        data-cy="gotItButton"
+                        isPrimary
+                      >
+                        <div>Got it</div>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/CreateWorkflow/ReliabilityScore/index.tsx:125]:
```suggestion
                        Got it
```

File: litmus-portal/frontend/src/views/CreateWorkflow/ReliabilityScore/index.tsx [hunk 5]
```diff
@@ -103,24 +103,29 @@ const ReliablityScore = () => {
...
-                    </ButtonFilled>
-                  </Center>
+                  <div>
+                    <ResultTable
+                      testValue={testWeights}
+                      testNames={testNames}
+                    />
+                  </div>
+                  <hr className={classes.horizontalLineResult} />
+                  <div className={classes.gotItBtn}>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/CreateWorkflow/ReliabilityScore/index.tsx:118]: This styling only has a margin-bottom. Can we add it along with ButtonFilled and make the code smaller!

**Review by S-ayanide — CHANGES_REQUESTED** [review]

**Review by S-ayanide — DISMISSED** [review]

**Review by amityt — DISMISSED** [review]

**Review by Saranya-jena — DISMISSED** [review]

**Review by S-ayanide — APPROVED** [review]

**Review by amityt — APPROVED** [review]

**Pull request merged**